packages/binding-wotfirestore/src/wotfirestore-client.ts

```diff
@@ -0,0 +1,214 @@
+    let timeoutId
+    const retContent: Content = await new Promise((resolve, reject) => {
+      subscribeFromFirestore(
+        this.firestore,
+        this.firestoreObservers,
+        actionResultTopic,
+        (err, content, resId) => {
+          console.debug('[debug] return action and unsubscribe')
+          console.debug(`[debug] reqId ${reqId}, resId ${resId}`)
+          if (reqId !== resId) {
+            // reqIdが一致しないため無視
+            return
+          }
+          unsubscribeFromFirestore(this.firestoreObservers, actionResultTopic)
+          clearTimeout(timeoutId)
+          if (err) {
+            console.error('[error] failed to get action result:', err)
+            reject(err)
+          } else {
+            resolve(content)
+          }
+        }
+      )
+      timeoutId = setTimeout(() => {
+        unsubscribeFromFirestore(this.firestoreObservers, actionResultTopic)
+        reject(new Error(`timeout error topic: ${pointerInfo.topic}`))
+      }, 10 * 1000) // timeout判定
+      // if not input was provided, set up an own body otherwise take input as body
+      if (content !== undefined) {
+        // アクション実行(結果は上記のCallbackに返る)
+        writeDataToFirestore(this.firestore, pointerInfo.topic, content, reqId)
+      } else {
+        // アクション実行(結果は上記のCallbackに返る)
+        writeDataToFirestore(
+          this.firestore,
+          pointerInfo.topic,
+          {
+            body: undefined,
+            type: ''
+          },
+          reqId
+        )
+      }
+    })
+    return retContent
+  }
+
+  public async unlinkResource(form: WoTFirestoreForm): Promise<any> {
+    const firestore = await initFirestore(this.fbConfig, this.firestore)
+    this.firestore = firestore
+    const pointerInfo = this.makePointerInfo(form)
+    unsubscribeFromFirestore(this.firestoreObservers, pointerInfo.topic)
+  }
+
+  public subscribeResource(
+    form: WoTFirestoreForm,
+    next: (value: any) => void,
+    error?: (error: any) => void,
+    complete?: () => void
+  ): any {
+    const pointerInfo = this.makePointerInfo(form)
+    // subscrbe for results
+    initFirestore(this.fbConfig, this.firestore)
+      .then((firestore) => {
+        this.firestore = firestore
+        subscribeFromFirestore(
+          this.firestore,
+          this.firestoreObservers,
+          pointerInfo.topic,
+          (err, content) => {
+            if (err) {
+              console.error('[error] failed to subscribe resource: ', err)
+              error(err)
+            } else {
+              next(content)
+            }
+          }
+        )
+      })
+      .catch((err) => {
+        console.error('[error] failed to init firestore: ', err)
+        error(err)
+      })
+  }
+
+  public start(): boolean {
+    return true
+  }
+
+  public stop(): boolean {
+    return true
+  }
+
+  public setSecurity(
+    metadata: Array<TD.SecurityScheme>,
+    credentials?: any
+  ): boolean {
+    // Firestoreにより通信路のセキュリティは確保
```

Comment:
what is this part?

Reply:
Now it always returns `true` because it does not use security. Is it correct to return `false` if it is not used?

Reply:
Ah I see. Then I am not sure. @danielpeintner what do you think?